app/src/main/kotlin/ExtendedStatelessSolver.kt

```diff
@@ -0,0 +1,40 @@
+import kotlin.math.*
+
+class ExtendedStatelessSolver<TState, TAction>(
+        mdp: MDP<TState, TAction>,
+        simulationDepthLimit: Int,
+        explorationConstant: Double,
+        rewardDiscountFactor: Double,
+        verbose: Boolean
+) : StatelessSolver<TState, TAction>(mdp, simulationDepthLimit, explorationConstant, rewardDiscountFactor, verbose) {
+
+    val rewardHistory = mutableListOf<Double>()
+
+    override fun constructTree(iterations: Int) {
+        for (i in 0..iterations) {
+            iterateStep()
+
+            val bestChild = _root.getChildren().maxByOrNull { c -> calculateUCT(c)}
+
+            println(_root.getChildren().toString())
+            println(bestChild)
+
+            val ns = bestChild?.n ?: continue
+            val explorationFactor = explorationConstant*sqrt(ln(i.toDouble())/ns)
+
+            rewardHistory.add(explorationFactor)
```

Comment:
Technically its not the reward history, but a history of `explorationFactor`

Reply:
I just used the previous naming, we could rename it.